team-manager/src/main/java/ua/knu/knudev/teammanager/service/QualificationImprovementService.java

```diff
@@ -0,0 +1,64 @@
+package ua.knu.knudev.teammanager.service;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import ua.knu.knudev.knudevcommon.constant.Expertise;
+import ua.knu.knudev.knudevcommon.constant.RecruitmentStatus;
+import ua.knu.knudev.teammanager.domain.Recruitment;
+import ua.knu.knudev.teammanager.repository.RecruitmentRepository;
+import ua.knu.knudev.teammanagerapi.api.QualificationImprovementApi;
+import ua.knu.knudev.teammanagerapi.dto.RecruitmentCloseRequestDto;
+import ua.knu.knudev.teammanagerapi.dto.RecruitmentOpenRequestDto;
+
+import java.time.LocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+public class QualificationImprovementService implements QualificationImprovementApi {
+
+    private final RecruitmentRepository recruitmentRepository;
+
+    @Override
+    public void initializeRecruitment(RecruitmentOpenRequestDto recruitmentOpenRequestDto) {
+        String recruitmentName = recruitmentOpenRequestDto.getRecruitmentName();
```

Comment:
Also check here if there isnt ongoing already opened recruitment with the same expertise